src/buffer_manager.ts

```diff
@@ -413,40 +428,46 @@ export class BufferManager implements Disposable {
     // #region Sync layout
 
     private onEditorLayoutChanged = async () => {
-        if (!this.syncEditorLayoutPromise) {
-            this.syncEditorLayoutPromise = new ManualPromise();
+        this.syncLayoutPromise = this.syncLayoutPromise ?? new ManualPromise();
+        this.isLayoutOutdated = true;
+        this.syncLayoutSource?.cancel();
+        this.syncLayoutSource = new CancellationTokenSource();
+        if (!this.isSyncingLayout) {
+            await this.syncEditorLayoutDebounced();
         }
-        this.syncEditorLayoutSource?.cancel();
-        this.syncEditorLayoutSource = new CancellationTokenSource();
-        await this.syncEditorLayoutDebounced(this.syncEditorLayoutSource.token);
     };
 
-    private syncEditorLayout = async (cancelToken: CancellationToken): Promise<void> => {
-        await this.cleanupWindowsAndBuffers();
-        if (cancelToken.isCancellationRequested) {
-            logger.debug(`Sync layout cancelled - syncVisibleEditors`);
-            return;
-        }
-        await this.syncVisibleEditors();
-        if (cancelToken.isCancellationRequested) {
-            logger.debug(`Sync layout cancelled - syncActiveEditor`);
-            return;
-        }
-        await this.syncActiveEditor();
-        if (cancelToken.isCancellationRequested) {
-            logger.debug(`Sync layout cancelled - resolve`);
-            return;
+    private syncEditorLayout = async (): Promise<void> => {
+        this.isSyncingLayout = true;
+        try {
+            while (this.isLayoutOutdated) {
+                this.isLayoutOutdated = false;
+                const token = this.syncLayoutSource?.token;
+
+                const visibleEditors = [...window.visibleTextEditors];
+                const activeEditor = window.activeTextEditor;
+
+                if (token?.isCancellationRequested) continue;
+                await this.cleanupWindowsAndBuffers(visibleEditors);
+
+                if (token?.isCancellationRequested) continue;
+                await this.syncVisibleEditors(visibleEditors);
+
+                if (token?.isCancellationRequested) continue;
+                await this.syncActiveEditor(activeEditor);
+            }
+        } catch (e) {
+            logger.error(`Error syncing layout: ${e}`);
```

Comment:
`%O` will often produce a better result than the default toString().
```suggestion
            logger.error(`Error syncing layout: %O`, e);
```

Reply:
Changed. Tip: The logger uses `inspect` to concatenate log information. So just pass it in as a separate parameter.